client/control/components/GroupMenu.js

```diff
@@ -0,0 +1,97 @@
+import React, { PropTypes as pt } from 'react';
+
+/**
+ * Simple menu component which displays groups of items
+ * under section headers/labels.
+ */
+export default class GroupMenu extends React.Component {
+  static propTypes = {
+    data: pt.array.isRequired,
+    onItemSelect: pt.func.isRequired,
+    searchText: pt.string,
+  };
+
+  /**
+   * Creates a click handler based on the
+   * group/option passed into it, basically
+   * an event handler factory.
+   *
+   * @param  {Object} group  Group which clicked option belongs to
+   * @param  {Object} option Option which was clicked
+   * @return {Function}      Wrapped handler
+   */
+  handleItemClick(group, option) {
+    return () => this.props.onItemSelect(group, option);
```

Comment:
>Alternatively, it might be simpler to cram it into the markup below...

I prefer to break out any anonymous functions that show up in templates (when possible). I think it helps keep the mental model in order when skimming code, and benefits refactoring. For instance, I just added handler caching without having to add any complexity to the template markup.

Anonymous functions are fired on every render too, which is a (very) slight perf hit, iirc it's better practice to not use anon functions in renders - but I could be wrong!